src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c

```diff
@@ -9,67 +13,514 @@ void EnKarebaba_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void EnKarebaba_Update(Actor* thisx, GlobalContext* globalCtx);
 void EnKarebaba_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-/*
+void EnKarebaba_DisableHitbox(EnKarebaba* this);
+void EnKarebaba_SetupGrow(EnKarebaba*);
+void EnKarebaba_SetupIdle(EnKarebaba*);
+void EnKarebaba_SetupAwaken(EnKarebaba* this);
+void EnKarebaba_SetupUpright(EnKarebaba* this);
+void EnKarebaba_SetupSpin(EnKarebaba* this);
+void EnKarebaba_SetupDying(EnKarebaba* this);
+void EnKarebaba_SetupDeadItemDrop(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_SetupRetract(EnKarebaba* this);
+void EnKarebaba_SetupWait(EnKarebaba* this);
+void EnKarebaba_SetupRegrow(EnKarebaba* this);
+void EnKarebaba_Grow(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Idle(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Awaken(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Upright(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Spin(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Dying(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_DeadItemDrop(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Retract(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Wait(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_Regrow(EnKarebaba* this, GlobalContext* globalCtx);
+void EnKarebaba_DrawCenterShadow(EnKarebaba* this, GlobalContext* globalCtx);
+
 const ActorInit En_Karebaba_InitVars = {
     ACTOR_EN_KAREBABA,
     ACTORTYPE_ENEMY,
-    FLAGS,
+    5,
     OBJECT_DEKUBABA,
     sizeof(EnKarebaba),
     (ActorFunc)EnKarebaba_Init,
     (ActorFunc)EnKarebaba_Destroy,
     (ActorFunc)EnKarebaba_Update,
-    (ActorFunc)EnKarebaba_Draw,
+    (ActorFunc)EnKarebaba_Draw
+};
+
+static ColliderCylinderInit karebabaHitbox = {
+    { 0xC, 0, 9, 0, 0x10, COLSHAPE_CYLINDER },
+    { 0, { 0x00000000, 0, 0 }, { ~0x00300000, 0, 0 }, 0, 1, 0 },
+    { 7, 25, 0, { 0, 0, 0 } }
+};
+
+static ColliderCylinderInit karebabaHitbox2 = {
+    { 0x0C, 0x11, 0, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0, { ~0x00300000, 0, 8 }, { 0x00000000, 0, 0 }, 9, 0, 1 },
+    { 4, 25, 0, { 0, 0, 0 }}
+};
+
+static CollisionCheckInfoInit karebabaLife = {
+    1, 15, 80, 254
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(unk_4C, 0x9C4, ICHAIN_CONTINUE),
+    ICHAIN_U8(unk_1F, 1, ICHAIN_CONTINUE),
+    ICHAIN_S8(naviEnemyId, 9, ICHAIN_STOP)
+};
+
+static Vec3f vecZero1 = { 0.0f, 0.0f, 0.0f };
+
+Color_RGBA8 colorBlack = { 0, 0, 0, 0 };
```

Comment:
also use `static` to restrict the symbols to this file (don't if it's used by a non-matching function's asm, but then use a less generic name)

and prefix data with `s` like for `sInitChain`